x-pack/plugins/data_enhanced/public/search/sessions_mgmt/lib/api.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { NotificationsStart } from 'kibana/public';
+import moment from 'moment';
+import { from, race, timer } from 'rxjs';
+import { mapTo, tap } from 'rxjs/operators';
+import type { SharePluginStart } from 'src/plugins/share/public';
+import { SessionsMgmtConfigSchema } from '../';
+import type { ISessionsClient } from '../../../../../../../src/plugins/data/public';
+import type { SearchSessionSavedObject } from '../../../../common';
+import { ACTION, STATUS, UISession } from '../../../../common/search/sessions_mgmt';
+
+type UrlGeneratorsStart = SharePluginStart['urlGenerators'];
+
+// Helper: factory for a function to map server objects to UI objects
+const mapToUISession = (
+  urls: UrlGeneratorsStart,
+  { expiresSoonWarning }: SessionsMgmtConfigSchema
+) => async (savedObject: SearchSessionSavedObject): Promise<UISession> => {
+  // Actions: always allow delete
+  const actions = [ACTION.DELETE];
```

Comment:
Looks like we're only ever showing `DELETE` in the actions menu. Should this be derived below?

Reply:
I updated the menu to include all relevant actions